src/main/java/seedu/duke/model/binarytree/BinaryTree.java

```diff
@@ -34,6 +34,16 @@ public T getRoot() {
         return arrayList.get(0);
     }
 
+    /**
+     * Returns T object at index of tree.
+     *
+     * @param idx Index in tree.
+     * @return T object.
+     */
+    public T getT(int idx) {
+        return arrayList.get(idx);
```

Comment:
We could assert that the logic will not call anything out of bounds?

Reply:
Good point. Should there be an assertion in `BooleanTemplate.setInput()` as well? Though it does seem we are using `DukeException` for `BinaryTree.insert()`. Maybe all as exceptions?

Reply:
Yup, sounds good!